### ISSUE TITLE
Weight Initialization PRNGKey

### ISSUE THREAD

**Issue by moabarar** [matthias-wright/flaxmodels#3]
Hi, 

First of all, you did a really good job on converting the stylegan2 implementation to jax. 
I am not sure if this is an issue or it was originally intended to design the module this way,  but one thing I noticed is that you pass the PRNG for the weight initialization as a parameter in the flax module. 

This means that initializing the module using flax.Module.init with different PRNG will result in the same weight initialization for both the generator and the discriminator. The only way to produce different initialization would be to pass different PRNGKeys at the creation of the StyleGAN module. 

Here's a minimal code that I made the demonstrates this:

```
# Create a StyleGAN2 Generator Flax.Module
G_model = stylegan2.generator.Generator(pretrained=None)

# Invoking this method will initialize the module based on the PRNGKey passed (i.e., g_rng)
def init_g(g_rng):
    z_shape = (4, 512)

    @jax.jit
    def _init(*args):
      return G_model.init(*args, train=True)

    variables = _init({'params': g_rng}, jnp.ones(z_shape, G_model.dtype))
    return variables['params'], variables['moving_stats'], variables['noise_consts']
```

Now initializing the module using the flax.Module.init will give the same params:

```
params_1, _, _ = init_g(jax.random.PRNGKey(10))
params_2, _, _ = init_g(jax.random.PRNGKey(58))
## The following returns true
jnp.alltrue( params_1['mapping_network']['LinearLayer_0']['weight'] == params_2['mapping_network']['LinearLayer_0']['weight'])
```

In order to initialize the module with different seeds, then you need to pass the PRNGKey explicitly at the creation of the Module.

If this was intended, then I think you also need to split the RNGs whenever the key is  passed to submodules. This will produce different random numbers for different weights.  Otherwise, creating the same layer twice (for example two ops.LinearLayer with the same hyperparameters, including the PRNGKey) will create the exact parameters for these layers.  

Thanks again for the hard work on making the module Jax accessible 👍

**Comment by matthias-wright** [comment]
Hi @moabarar! Thank you for pointing this out, this was definitely not intended.
I have now fixed this issue (see 5adfc79f6dd2ce7e750b66d2a9f41fa5fb1a96c6).

I replaced `ops.get_weight` with `ops.get_weight_init` and `ops.get_bias_init`.
In `ops.get_weight`, the parameters were actually initialized using the PRNG that was the same for all sub-modules as you pointed out. 
Both `ops.get_weight_init` and `ops.get_bias_init` now return a callable, which are passed to `flax.linen.Module.param`. The callables take in a PRNG as argument, the way it is also done in all the other modules in `flax.linen`.

I tried it with your example and now the weights are different for different PRNGs.
Thanks again!
